src/uwtools/drivers/jedi.py

```diff
@@ -0,0 +1,134 @@
+"""
+A driver for the jedi component.
+"""
+
+import logging
+from datetime import datetime
+from pathlib import Path
+
+from iotaa import asset, dryrun, run, task, tasks
+
+from uwtools.config.formats.yaml import YAMLConfig
+from uwtools.drivers.driver import Driver
+from uwtools.strings import STR
+from uwtools.utils.tasks import file, filecopy, symlink
+
+
+class JEDI(Driver):
+    """
+    A driver for the JEDI component.
+    """
+
+    def __init__(self, config: Path, cycle: datetime, dry_run: bool = False, batch: bool = False):
+        """
+        The driver.
+
+        :param config: Path to config file.
+        :param cycle: The forecast cycle.
+        :param dry_run: Run in dry-run mode?
+        :param batch: Run component via the batch system?
+        """
+        super().__init__(config=config, dry_run=dry_run, batch=batch, cycle=cycle)
+        if self._dry_run:
+            dryrun()
+        self._cycle = cycle
+
+    # Workflow tasks
+
+    @tasks
+    def files_copied(self):
+        """
+        Files copied for run.
+        """
+        yield self._taskname("files copied")
+        yield [
+            filecopy(src=Path(src), dst=self._rundir / dst)
+            for dst, src in self._driver_config.get("files_to_copy", {}).items()
+        ]
+
+    @tasks
+    def files_linked(self):
+        """
+        Files linked for run.
+        """
+        yield self._taskname("files linked")
+        yield [
+            symlink(target=Path(target), linkname=self._rundir / linkname)
+            for linkname, target in self._driver_config.get("files_to_link", {}).items()
+        ]
+
+    @tasks
+    def provisioned_run_directory(self):
+        """
+        Run directory provisioned with all required content.
+        """
+        yield self._taskname("provisioned run directory")
+        yield [
+            self.files_copied(),
+            self.files_linked(),
+            self.configuration_file(),
+            self.runscript(),
+        ]
+
+    @task
+    def runscript(self):
+        """
+        The runscript.
+        """
+        path = self._runscript_path
+        yield self._taskname(path.name)
+        yield asset(path, path.is_file)
+        yield None
+        self._write_runscript(path=path, envvars={})
+
+    @task
+    def validate_only(self):
+        """
+        Validate config.
+        """
+        taskname = self._taskname("validate_only")
+        yield taskname
+        a = asset(None, lambda: False)
+        yield a
+        path = Path(self._driver_config["configuration_file"]["base_file"])
```

Comment:
```suggestion
        base_file = Path(self._driver_config["configuration_file"]["base_file"])
```
Other lines will have to be updated to match. Since `executable` is also a path, I find it odd to find this one called `path`. This may also change depending on what happens with the comment I made above about validating the `configuration_file` file instead of the `base_file`.